src/Utilities/MoistThermodynamics/src/MoistThermodynamics.jl

```diff
@@ -462,49 +784,212 @@ function saturation_adjustment(e_int, ρ, q_tot, T_init = T_triple)
     return air_temperature(e_int, q_tot, q_liq, q_ice)
   end
 end
+function saturation_adjustment_q_t_θ_l(θ_liq::DT, q_tot::DT, ρ::DT, p::DT) where DT
+  T_1 = θ_liq*(p/DT(MSLP))^(DT(R_d)/DT(cp_d))
+  qv_star = saturation_shum(T_1, ρ)
+  if (q_tot <= qv_star) # If not saturated
+    return T_1
+  else  # If not saturated, iterate
+    T_2 = T_1 + (q_tot - qv_star) * latent_heat_vapor(T_1) /((1-q_tot)*DT(cp_d) + qv_star * DT(cp_v))
```

Comment:
Use saturation excess function and partitioning into liquid and ice, then the correct gas constants for moist air.